### PR TITLE
Metric Registry: Auto convert getter name into a property name

### PR DESCRIPTION
**Pull request by jerrinot**
Example
```java
@Probe
public int getFoo() {
 [...]
}
```

Now generates a probe named `foo` instead of `getFoo`

This increases usefulness of implicit probe names and does not force
you to use explicit names only to remove the get- prefix.

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/util/StringUtil.java [hunk 1]
```diff
@@ -247,4 +251,38 @@ public static int lastIndexOf(String input, char ch) {
...
+     * It's written defensively, when output is not a getter then it
+     * return the original name.
+     *
+     * It converters name starting with the get- prefix only. When a getter
+     * starts with is- prefix (=boolean) then it does not convert it.
+     *
+     * @param getterName
+     * @return property matching the given getter
+     */
+    public static String getterIntoProperty(String getterName) {
```
Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/util/StringUtil.java]: this is where the magic happens

File: hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MethodProbe.java [hunk 2]
```diff
@@ -63,8 +64,10 @@ void register(MetricsRegistryImpl metricsRegistry, Object source, String namePre
     }
 
     private String getName(String namePrefix) {
-        String name = method.getName();
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MethodProbe.java]: we need to make sure that we are not going to generate litter when the diagnostics writes to e.g. file. I don't have the logic of this part in my head at the moment.

Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/MethodProbe.java]: `getName()` is private and is used during a probe registration only.

File: hazelcast/src/main/java/com/hazelcast/util/StringUtil.java [hunk 3]
```diff
@@ -247,4 +253,38 @@ public static int lastIndexOf(String input, char ch) {
...
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/util/StringUtil.java]: The name is not revealing the intent; I can't guess based on the name of this method what the purpose is.

Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/util/StringUtil.java]: my original choice was something like `stringGetterPrefix()`. but there is more going on - e.g. conversion to lower case. 

bear in mind this is located in a class `StringUtil`. do you have a better name?

Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/util/StringUtil.java]: toPropertyName? And the argument I would call 'getMethodName'

Comment (jerrinot) [inline hazelcast/src/main/java/com/hazelcast/util/StringUtil.java]: could do

**Comment by pveentjer** [comment]
:+1:

**Pull request merged**